int16 reset
---
bool done